### PR TITLE
Roadmap update per 2019-10-30 meeting

### PR DESCRIPTION
**Pull request by GeoffreyBooth**
Updated per our consensuses reached in today’s meeting.

https://github.com/nodejs/modules/pull/416/files?short_path=efa67e9#diff-efa67e9109ca1a93520741c28eb913c9

### CODE REVIEW

File: doc/plan-for-new-modules-implementation.md [hunk 1]
```diff
@@ -103,44 +103,45 @@ Phase 3 improves user experience and extends the MVP. Phase 3 is malleable based
...
+  - Issue raised in: https://github.com/nodejs/node/pull/29426.
+  - Landed in https://github.com/nodejs/node/pull/29737 and shipped in 12.12.0.
+
+* Shortcut to resolve to package root.
+  - Proposal: [Package `"name"` Resolution Proposal](https://github.com/guybedford/package-name-resolution).
+  - Discussion: https://github.com/nodejs/modules/issues/306.
+  - Landed in https://github.com/nodejs/node/pull/29327.
+
+* Dual CommonJS/ESM packages: Support packages with both CommonJS and ESM sources that can be used in either environment.
+  - At time of unflagging: `"main"` points to exactly one file, and all file extensions are mandatory (by default), so there is no possibility of an `import` specifier pointing to different files in ESM versus CommonJS; unless `--experimental-conditional-exports` is used. Without that flag, dual packages must provide secondary entry point via a path, e.g. `'pkg/module'` or `'pkg/commonjs'`. With `--experimental-conditional-exports`, paths within the `package.json` `"exports"` block can have separate entry points per environment.
```
Comment (MylesBorins) [inline doc/plan-for-new-modules-implementation.md]: "extensions are mandatory" is not entirely accurate. "Full filename" seems more accurate to me

Also think that information regarding the experimental flag should likely be a separate bullet

File: doc/plan-for-new-modules-implementation.md [hunk 2]
```diff
@@ -103,44 +103,45 @@ Phase 3 improves user experience and extends the MVP. Phase 3 is malleable based
...
+  - PR for conditional exports: https://github.com/nodejs/node/pull/29978.
+  - PR for additional docs for best practices on writing dual packages with conditional exports: https://github.com/nodejs/node/pull/30051. [Formatted version](https://github.com/nodejs/node/pull/30051/files?short_path=8e67f40#diff-8e67f407bc32a0569e25d7ecaff6e494); start at the “Dual CommonJS/ES Module Packages” heading.
+  - Some members of the group expressed hesitation about conditional exports and want to explore other potential solutions for dual packages. If no alternative proposal reaches consensus before January 2020, the `--experimental-conditional-exports` flag will be dropped and conditional exports will ship.
```
Comment (MylesBorins) [inline doc/plan-for-new-modules-implementation.md]: "before January 2020", probably better to frame this as "end of January 2020". Might also be worth mentioning that flag could drop sooner if consensus is reached on team.

File: doc/plan-for-new-modules-implementation.md (hunk 1 above)
Comment (ljharb) [reply doc/plan-for-new-modules-implementation.md]: Also, with exports, it’s trivial to make extensions *forbidden* - so myles’ suggestion plus a comment about exports probably would be better.

Comment (GeoffreyBooth) [reply doc/plan-for-new-modules-implementation.md]: FWIW this didn't change in this PR, but I'm fine with changing it.

**Comment by GeoffreyBooth** [comment]
Revised, @MylesBorins please take a look.

File: doc/plan-for-new-modules-implementation.md [hunk 3]
```diff
@@ -103,44 +103,47 @@ Phase 3 improves user experience and extends the MVP. Phase 3 is malleable based
...
+
+* Finalize behavior of `import` of CommonJS files and packages.
+  - Overview: https://github.com/nodejs/modules/issues/264.
+  - At time of unflagging: `import` only the CommonJS default export, so `import _ from 'cjs-pkg'` but not `import { shuffle } from 'cjs-pkg'`.
+  - [Conditional exports](https://github.com/nodejs/node/pull/29978) allows creating an ESM wrapper to provide named exports of an otherwise all-CommonJS package; see [heading “Approach #1: Use an ES Module Wrapper.”](https://github.com/nodejs/node/pull/30051/files?short_path=8e67f40#diff-8e67f407bc32a0569e25d7ecaff6e494)
+  - **Status**: No further improvements expected.
+
+* Dual CommonJS/ESM packages: Support packages with both CommonJS and ESM sources that can be used in either environment.
+  - At time of unflagging: `"main"` points to exactly one file, and full filenames are required (by default), so there is no possibility of an `import` specifier pointing to different files in ESM versus CommonJS; unless `--experimental-dual-resolution` is used (see next bullet). Without that flag, dual packages must provide secondary entry point via a path, e.g. `'pkg/module'` or `'pkg/commonjs'`.
```
Comment (MylesBorins) [inline doc/plan-for-new-modules-implementation.md]: should this mention `pkg.exports['.']`?

**Pull request merged**